packages/client/tests/functional/referential-integrity/tests.ts

```diff
@@ -0,0 +1,677 @@
+          data: {
+            id: '2',
+            profile: {
+              create: { id: '2' },
+            },
+          },
+        })
+
+        const user1WithNewId = await prisma[userModel].update({
+          where: { id: '1' },
+          data: {
+            id: '3',
+          },
+        })
+        expect(user1WithNewId).toEqual({
+          id: '3',
+        })
+
+        // Checks
+        const profileNull = await prisma[profileModel].findFirst({
+          where: { userId: '1' },
+        })
+        expect(profileNull).toEqual(null)
+        const profiles = await prisma[profileModel].findMany({
+          orderBy: [{ id: 'asc' }],
+        })
+        expect(profiles).toEqual([
+          {
+            id: '1',
+            userId: '3',
+          },
+          {
+            id: '2',
+            userId: '2',
+          },
+        ])
+      })
+
+      test('[update] parent id with existing id should throw', async () => {
+        await prisma[userModel].create({
+          data: {
+            id: '1',
+            profile: {
+              create: { id: '1' },
+            },
+          },
+        })
+        await prisma[userModel].create({
+          data: {
+            id: '2',
+            profile: {
+              create: { id: '2' },
+            },
+          },
+        })
+
+        await expect(
+          prisma[userModel].update({
+            where: { id: '1' },
+            data: {
+              id: '2', // existing id
+            },
+          }),
+        ).rejects.toThrowError('Unique constraint failed on the fields: (`id`)')
+      })
+
+      test('[update] child id with non-existing id should succeed', async () => {
+        await prisma[userModel].create({
+          data: {
+            id: '1',
+            profile: {
+              create: { id: '1' },
+            },
+          },
+        })
+        await prisma[userModel].create({
+          data: {
+            id: '2',
+            profile: {
+              create: { id: '2' },
+            },
+          },
+        })
+
+        const profile1WithNewId = await prisma[profileModel].update({
+          where: { id: '1' },
+          data: {
+            id: '3',
+          },
+        })
+        expect(profile1WithNewId).toEqual({
+          id: '3',
+          userId: '1',
+        })
+
+        const user1WithUpdatedProfile = await prisma[userModel].findUniqueOrThrow({
+          where: { id: '1' },
+          include: { profile: true },
+        })
+        expect(user1WithUpdatedProfile).toEqual({
+          id: '1',
+          profile: {
+            id: '3',
+            userId: '1',
+          },
+        })
+
+        const profiles = await prisma[profileModel].findMany({
+          orderBy: [{ id: 'asc' }],
+        })
+        expect(profiles).toEqual([
+          {
+            id: '2',
+            userId: '2',
+          },
+          {
+            id: '3',
+            userId: '1',
+          },
+        ])
```

Comment:
Using `findMany()` for each model, would simplify assertions in all tests, and show/verify what we expect to see for all the items (findUniqueOrThrow only verifies 1 item)